lib/spack/llnl/util/lock.py

```diff
@@ -386,8 +386,17 @@ def _ensure_parent_directory(self):
         try:
             os.makedirs(parent)
         except OSError as e:
-            # makedirs can fail when diretory already exists.
-            if not (e.errno == errno.EEXIST and os.path.isdir(parent) or e.errno == errno.EISDIR):
+            if e.errno == errno.EISDIR:
+                # Directory already exists
+                pass
+            elif os.path.isdir(parent):
+                # Directory already exists
+                if e.errno != errno.EEXIST:
+                    tty.warn(
+                        "Got OSError when trying to create directory "
+                        "at '{!s}' that already exists: {!r}".format(parent, e)
+                    )
+            else:
```

Comment:
```suggestion
            # os.makedirs can fail in a number of ways when the directory already exists.
            # With EISDIR, we know it exists, and others like EEXIST, EACCES, and EROFS
            # are fine if we ensure that the directory exists.
            # Python 3 allows an exist_ok parameter and ignores any OSError as long as
            # the directory exists.
            if not (e.errno == errno.EISDIR or os.path.isdir(parent)):
```

I don't think we need to warn if we're doing basically what cpython would do in Python 3.